packages/teams-js/src/public/pages.ts

```diff
@@ -1096,4 +1103,117 @@ export namespace pages {
         : false;
     }
   }
+
+  /**
+   * Provides APIs to interact with the button in hosts
+   *
+   * @beta
+   */
+  export namespace responseButton {
+    /**
+     * The required information that the app needs to let host apps know.
+     *
+     * @beta
+     */
+    export interface ResponseInfo {
+      /**
+       * The unique ID helps the host and apps identify which button it refers to.
+       *
+       * @beta
+       */
+      responseId: string;
+      /**
+       * Id of File on OneDrive
+       *
+       * @beta
+       */
+      oneDriveFileId: string;
```

Comment:
oneDriveFileId, messageId, and conversationId are probably values that should be passed into the app as part of the ActionInfo object in the context that is defined in `interfaces.ts`:

```typescript
/**
 * Information common to all actions
 *
 * @beta
 */
export interface ActionInfo {
  /**
   * Maps to the action id supplied inside the manifest
   */
  actionId: string;
  /**
   * Array of corresponding action objects
   */
  actionObjects: BaseActionObject<ActionObjectType>[];
}
```

In ActionInfo, the `actionObjects` array contains one or more items than an action has been taken on. In this case, there will be at least one item representing the file that the action has been taken on:

```typescript
/**
 * Stores information needed to represent M365 Content stored
 * in OneDrive or Sharepoint
 *
 * @beta
 */
export interface M365ContentAction extends BaseActionObject<ActionObjectType.M365Content> {
  /**
   * Only office content IDs are passed to the app. Apps should use these ids
   * to query the Microsoft graph for more details.
   */
  itemId: string;
  /** Represents an optional secondary identifier for an action in a Microsoft 365 content item. */
  secondaryId?: SecondaryId;
}
```

In this case, `M365ContentAction.itemId` should be the oneDriveId of the file, and the `secondaryId` is an optional second id that may be needed in order to retrieve that file from the MS Graph (for example, the host can pass down a `driveId` if one is interact with the file in the graph). In general, this item being an `M365ContentAction` is also used as a signal to app developers that they can use the information specified here to download the file from a particular MS Graph Endpoint, detailed here: https://learn.microsoft.com/en-us/graph/api/resources/onedrive?view=graph-rest-1.0

If I remember correctly, the host will be taking this file and moving it to OneDrive before opening the app, so the host passing in an M365ContentAction still feels correct. However, based on what you've specified below, the host will also need to pass in a messageId and conversationId in the M365ContentAction so that the developer can pass them back to the host in order to initialize the ResponseButton.

If I have all of that right, M365ContentAction should probably have some sort of new optional property named something like 'originalSource` or `originalOwner` that contains that information. If you had that, then `ResponseInfo` could be simplified a lot to look like this:

```typescript
/**
     * The required information that the app needs to let host apps know.
     *
     * @beta
     */
    export interface ResponseInfo {
      /**
       * The unique ID helps the host and apps identify which button it refers to.
       *
       * @beta
       */
      responseId: string;
      /**
       * This should be one of the items passed to the app in the ActionInfo. Please pass the item back to initialize the
       * ResponseButton for the corresponding item that an action was taken on. Note just for maggie: this text is not final, more an 
       * explanation for you. You may want to change it to make it more meaningful for external developers
       * @beta
       */
      item: BaseActionObject<ActionObjectType>;
    }
```

Reply:
Thanks for your suggestion, those are all really helpful. 
After talking with PMs @irenewang101 and @Deepthi, we've decided to add originalSource under M365ContentAction. This allows every actionObject to have its own originalSource, making it easier to extend for our hosts.